src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/FileBasedProgramsProjectSystem.cs

```diff
@@ -62,12 +56,16 @@ public FileBasedProgramsProjectSystem(
         _logger = loggerFactory.CreateLogger<FileBasedProgramsProjectSystem>();
         _metadataAsSourceFileService = metadataAsSourceFileService;
         _projectXmlProvider = projectXmlProvider;
+        _workspaceFactory = workspaceFactory;
     }
 
-    public Workspace Workspace => ProjectFactory.Workspace;
-
     private string GetDocumentFilePath(DocumentUri uri) => uri.ParsedUri is { } parsedUri ? ProtocolConversions.GetDocumentFilePathFromUri(parsedUri) : uri.UriString;
 
+    public async ValueTask<bool> IsProvidedDocumentAsync(TextDocument document)
```

Comment:
this name isn't that clear to me.  like "provided document" doesn't really convey something to me.

Reply:
I don't like it either. I'm also not even really sure why this method has to exist -- we only are using it to gate calling TryRemove....which can just check the same thing again? This would be good to discuss during the call on Monday.